test/unit/reducer/tests/test-reducer-constructors.hpp

```diff
@@ -0,0 +1,194 @@
+  RAJA::ReduceMinLoc<ReducePolicy, NumericType> reduce_minloc;
+  RAJA::ReduceMaxLoc<ReducePolicy, NumericType> reduce_maxloc;
+
+  RAJA::ReduceMinLoc<ReducePolicy, NumericType, RAJA::tuple<RAJA::Index_type, RAJA::Index_type>> reduce_minloctup;
+  RAJA::ReduceMaxLoc<ReducePolicy, NumericType, RAJA::tuple<RAJA::Index_type, RAJA::Index_type>> reduce_maxloctup;
+
+  ASSERT_EQ((NumericType)reduce_sum.get(), NumericType());
+  ASSERT_EQ((NumericType)reduce_min.get(), NumericType());
+  ASSERT_EQ((NumericType)reduce_max.get(), NumericType());
+
+  ASSERT_EQ((NumericType)reduce_minloc.get(), NumericType());
+  ASSERT_EQ((NumericType)reduce_maxloc.get(), NumericType());
+  ASSERT_EQ((RAJA::Index_type)reduce_minloc.getLoc(), RAJA::Index_type());
+  ASSERT_EQ((RAJA::Index_type)reduce_maxloc.getLoc(), RAJA::Index_type());
+
+  ASSERT_EQ((NumericType)reduce_minloctup.get(), NumericType());
+  ASSERT_EQ((NumericType)reduce_maxloctup.get(), NumericType());
+  ASSERT_EQ((RAJA::Index_type)(RAJA::get<0>(reduce_minloctup.getLoc())), RAJA::Index_type());
+  ASSERT_EQ((RAJA::Index_type)(RAJA::get<1>(reduce_minloctup.getLoc())), RAJA::Index_type());
+  ASSERT_EQ((RAJA::Index_type)(RAJA::get<0>(reduce_maxloctup.getLoc())), RAJA::Index_type());
+  ASSERT_EQ((RAJA::Index_type)(RAJA::get<1>(reduce_maxloctup.getLoc())), RAJA::Index_type());
+}
+
+TYPED_TEST_P(ReducerBasicConstructorUnitTest, BasicReducerConstructor)
+{
+  using ReducePolicy = typename camp::at<TypeParam, camp::num<0>>::type;
+  using NumericType = typename camp::at<TypeParam, camp::num<1>>::type;
+
+  testReducerConstructor< ReducePolicy, NumericType >();
+}
+
+template  < typename ReducePolicy,
+            typename NumericType,
+            typename ResourceType,
+            typename ForOnePol >
+typename  std::enable_if< // Host policy sets value.
+            std::is_same<ResourceType, camp::resources::Host>::value
+            #if defined(RAJA_ENABLE_TARGET_OPENMP)
+            || std::is_same<ReducePolicy, camp::resources::Omp>::value
+            #endif
+          >::type
+exec_dispatcher( NumericType * theVal, NumericType * initVal )
+{
+  theVal[0] = initVal[0];
+}
+
+template  < typename ReducePolicy,
+            typename NumericType,
+            typename ResourceType,
+            typename ForOnePol >
+typename  std::enable_if< // GPU policy sets value.
+            !std::is_same<ResourceType, camp::resources::Host>::value
+            #if defined(RAJA_ENABLE_CUDA)
+            && std::is_same<ResourceType, camp::resources::Cuda>::value
+            #endif
+            #if defined(RAJA_ENABLE_HIP)
+            && std::is_same<ResourceType, camp::resources::Hip>::value
+            #endif
+          >::type
+exec_dispatcher( NumericType * theVal, NumericType * initVal )
+{
+  #if defined(RAJA_ENABLE_CUDA) || defined(RAJA_ENABLE_HIP)
+  forone_pol<ForOnePol>( [=] __device__ () {
+                        theVal[0] = initVal[0];
+                 });
+  #endif
+}
+
```

Comment:
I wish there was a better way, but this is probably what we should do.
Maybe split off the test in the enable if into its own class to make this less ugly and let us reuse this pattern.

Reply:
I agree. I missed this. Thanks for the comment @MrBurmark

Reply:
@MrBurmark @rhornung67 Ok, I created a couple classes in test-reducer-utils.hpp to make this cleaner. Let me know what you think.

Reply:
It is much cleaner, and the platform makes it clear what's going on. It feels like the forone policy itself has this information, so it may be worthwhile to avoid the extra template param add the platform as part of the forone policy traits.

Reply:
Good point! I've moved the RunOn... types into forone, and used inheritance to effect is_base_of.